### ISSUE TITLE
Missing compiler warning for not awaiting Task produced by delegate?

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#76721]
**Version Used**: main

**Steps to Reproduce**:

```csharp
using System;
using System.Threading.Tasks;

internal delegate Task MyDelegate();

public class TestClass
{
    public void M()
    {
        MyDelegate c = async () =>
        {
            await Task.Delay(1);
            throw new Exception();
        };
        // Expecting to see: CS4014
        // warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
        // But no warning is produced.
        c();
    }
}
```

If I switch to a local function instead of a delegate, the warning is correctly reported.

```csharp
using System;
using System.Threading.Tasks;

internal delegate Task MyDelegate();


public class TestClass
{
    public void M()
    {
        async Task c()
        {
            await Task.Delay(1);
            throw new Exception();
        }
        c();
    }
}
```

**Comment by jaredpar** [comment]
The compiler is pretty conservative on when it will warn for missing `await` on a task-like type. The goal is to avoid introducing lots of warnings into code which is not `async` aware but does use `Task` APIs; originally a nod to compat with existing code. I'm unsure why the C# team decided to not warn on the first example in C# 5 but at this point there is very little chance that we would change this behavior.

**Comment by Youssef1313** [comment]
@jaredpar Not even in a warning wave?

**Comment by jaredpar** [comment]
Not for this. It's a pretty established pattern and code bases are sensitive to this warning.

**Comment by Youssef1313** [comment]
> pretty established pattern

I'm surprised that people ended up doing it intentionally 😮 Exceptions could end up not being observed which is pretty bad. For context, we have had similar cases of `Assert`s being swallowed due to similar patterns.

FYI @Evangelink. I noticed it while working on https://github.com/microsoft/testfx/pull/4640. The case in this issue isn't going to be caught by the analyzer I did as the delegate itself is `async Task`, but is called without `await`.
